### PR TITLE
chore: migrate to main branch

### LINKED ISSUE #430: Migrate to main branch
Contributing to googleapis/google-cloud-python#10579, switching the default branch to main branch. Also replacing all occurrences of master to main.

### PR DESCRIPTION
**Pull request by dandhlee**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-firestore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #430  🦕

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -16,7 +16,7 @@ including Cloud Functions.
 -  `Client Library Documentation`_
 
 .. |GA| image:: https://img.shields.io/badge/support-GA-gold.svg
-   :target: https://github.com/googleapis/google-cloud-python/blob/master/README.rst#general-availability
+   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#general-availability
```
Comment (parthea) [inline README.rst:19]: I get 404 with the new link

File: samples/AUTHORING_GUIDE.md [hunk 2]
```diff
@@ -1 +1 @@
-See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md
\ No newline at end of file
+See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md
```
Comment (parthea) [inline samples/AUTHORING_GUIDE.md:1]: I get 404 with the new link

**Review by parthea — COMMENTED** [review]
Minor observations, otherwise LGTM.

**Comment by dandhlee** [comment]
Please take a look again!

File: README.rst (hunk 1 above)
Comment (dandhlee) [reply README.rst]: Done.

File: samples/AUTHORING_GUIDE.md (hunk 2 above)
Comment (dandhlee) [reply samples/AUTHORING_GUIDE.md]: Done.

**Comment by tseaver** [comment]
The link into `google-cloud-python` (for the GA explanation) is still broken, but shouldn't block this PR:  I'm assuming we're waiting to migrate that repo until the end?

**Review by parthea — APPROVED** [review]

**Pull request merged**